### PR TITLE
feat(cli): Add settings.max_width and settings.max_col_width

### PR DESCRIPTION
**Pull request by TCeason**
```
root@localhost> .max_col_width 5;
root@localhost> .max_display_rows 4;
root@localhost> .max_width 40;

root@localhost> desc db.hits;

DESC db.hits

┌───────────────────────────────────────────┐
│   Fi...   │ Ty... │ Nu... │ De... │ Ex... │
│   St...   │ St... │ St... │ St... │ St... │
├───────────┼───────┼───────┼───────┼───────┤
│ wa...     │ BI... │ NO    │ 0     │       │
│ ja...     │ SM... │ NO    │ 0     │       │
│ ·         │ ·     │ ·     │ ·     │ ·     │
│ ·         │ ·     │ ·     │ ·     │ ·     │
│ ·         │ ·     │ ·     │ ·     │ ·     │
│ ur...     │ BI... │ NO    │ 0     │       │
│ clid      │ INT   │ NO    │ 0     │       │
│ 105 rows  │       │       │       │       │
│ (4 shown) │       │       │       │       │
└───────────────────────────────────────────┘
105 rows in 0.035 sec. Processed 0 rows, 0B (0 rows/s, 0B/s)

root@localhost> select * from db.hits;

SELECT
  *
FROM
  db.hits

┌─────────────────────────────────────────────────────────┐
│   wa...   │ ja... │ ti... │ ··· │ re... │ ur... │ cl... │
│   In...   │ In... │ St... │     │ In... │ In... │ In... │
├───────────┼───────┼───────┼─────┼───────┼───────┼───────┤
│     91... │     0 │       │ ... │ -2... │ -8... │     0 │
│     81... │     0 │       │ ... │ -2... │ -8... │     0 │
│         · │     · │ ·     │  ·  │     · │     · │     · │
│         · │     · │ ·     │  ·  │     · │     · │     · │
│         · │     · │ ·     │  ·  │     · │     · │     · │
│     66... │     0 │       │ ... │ -2... │ 49... │     0 │
│     89... │     0 │       │ ... │ -2... │ -8... │     0 │
│   10 rows │       │       │     │       │       │       │
│ (4 shown) │       │       │     │       │       │       │
└─────────────────────────────────────────────────────────┘
10 rows in 0.047 sec. Processed 10 rows, 10B (214.17 rows/s, 70.25 KiB/s)
```

### CODE REVIEW

File: cli/src/config.rs [hunk 1]
```diff
@@ -158,6 +164,8 @@ impl Default for Settings {
             output_format: OutputFormat::Table,
             show_progress: false,
             max_display_rows: 40,
+            max_col_width: 0,
+            max_width: 0,
```
Comment (sundy-li) [inline cli/src/config.rs:168]: can we set it to be terminal width ?

File: cli/src/config.rs [hunk 2]
```diff
@@ -158,6 +164,8 @@ impl Default for Settings {
             output_format: OutputFormat::Table,
             show_progress: false,
             max_display_rows: 40,
+            max_col_width: 0,
```
Comment (sundy-li) [inline cli/src/config.rs:167]: max_col_width could default to be 20

File: cli/src/display.rs [hunk 3]
```diff
@@ -363,6 +517,68 @@ fn create_table(schema: SchemaRef, results: &[Row], max_rows: usize) -> Result<T
...
+                let width = widths[i];
+                let field_name = if field.name.len() + 3 > *max_col_width {
+                    field.name[0..max_col_width - 3].to_string() + "..."
+                } else if field.name.len() + 3 > width {
+                    field.name[0..width - 3].to_string() + "..."
+                } else {
+                    field.name.to_string()
+                };
+
+                let field_type = if field.data_type.to_string().len() + 3 > *max_col_width {
```
Comment (sundy-li) [inline cli/src/display.rs:559]: let s = field.data_type.to_string();
if ...

File: cli/src/config.rs (hunk 1 above)
Comment (TCeason) [reply cli/src/config.rs:168]: Duckdb is this: if max width is 0, will use terminal width. But I don't know how to get the terminal width in rust.

Comment (TCeason) [reply cli/src/config.rs:168]: Done

File: cli/src/config.rs (hunk 2 above)
Comment (TCeason) [reply cli/src/config.rs]: Done

File: cli/src/display.rs (hunk 3 above)
Comment (TCeason) [reply cli/src/display.rs]: Done

File: cli/src/display.rs [hunk 4]
```diff
@@ -279,28 +286,125 @@ fn display_read_progress(pb: Option<ProgressBar>, current: &QueryProgress) -> Pr
...
+    max_width: usize,
+    max_col_width: usize,
+) -> (Vec<usize>, Vec<i32>) {
+    let column_count = schema.fields().len();
+    let mut widths = Vec::with_capacity(column_count);
+    let mut total_length = 1;
+
+    for field in schema.fields() {
+        // head_name = field_name + "\n" + field_data_type
+        let col_length = field.name.len() + 1 + field.data_type.to_string().len();
```
Comment (TCeason) [inline cli/src/display.rs:301]: image

Comment (TCeason) [reply cli/src/display.rs]: Fixed.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**